tests/pytest-pypi/pytest_pypi/app.py

```diff
@@ -4,51 +4,54 @@
 import requests
 from flask import Flask, redirect, abort, render_template, send_file, jsonify
 
-PYPI_VENDOR_DIR = os.environ.get('PYPI_VENDOR_DIR', './pypi')
-PYPI_VENDOR_DIR = os.path.abspath(PYPI_VENDOR_DIR)
-
 app = Flask(__name__)
 session = requests.Session()
 
 packages = {}
 
 
 class Package(object):
-    """docstring for Package"""
+    """Package represents a collection of releases from one or more directories"""
 
     def __init__(self, name):
         super(Package, self).__init__()
         self.name = name
-        self._releases = []
+        self.releases = {}
+        self._package_dirs = set()
 
     @property
-    def releases(self):
-        r = []
-        for release in self._releases:
-            release = release[len(PYPI_VENDOR_DIR):].replace('\\', '/')
-            r.append(release)
-        return r
+    def json(self):
+        for path in self._package_dirs:
+            try:
+                with open(os.path.join(path, 'api.json')) as f:
+                    return json.load(f)
+            except FileNotFoundError:
+                pass
 
     def __repr__(self):
         return "<Package name={0!r} releases={1!r}".format(self.name, len(self.releases))
 
     def add_release(self, path_to_binary):
-        self._releases.append(path_to_binary)
-
-
-def prepare_packages():
-    for root, dirs, files in os.walk(os.path.abspath(PYPI_VENDOR_DIR)):
+        path_to_binary = os.path.abspath(path_to_binary)
+        path, release = os.path.split(path_to_binary)
+        self.releases[release] = path_to_binary
+        self._package_dirs.add(path)
+
+
+def prepare_packages(path):
+    """Add packages in path to the registry."""
```

Comment:
These changes are so straightforward and so much more readable.  I've looked at tackling a lot of this before, basically spent about 5 minutes on it and ducked out because I broke everything... Thank you for the cleanup, this is very easy to follow and the changes are both logical and clear.